[FEAT/#48][FIX/#60] Custom ImageButton 제작, Coil 라이브러리 수정

TaewoongR opened the pull request:
- closed #48 
- closed #60

## *📍 Work Description*

- Custom ImageButton 제작 (초기 화면은 이미지 아이콘과 문자열이 수직으로 나열되어 있습니다.)
- 코일을 결합하여 사용 및 에뮬레이터 테스트

## *📸 Screenshot*

https://github.com/user-attachments/assets/3a5e97cf-337b-4da6-a5d6-50c26fc04e13

<img src="https://github.com/user-attachments/assets/4c70863f-1a1c-48a4-9932-1e6e9ee377af" width=600>

## *📢 To Reviewers*
- 이 컴포넌트의 용도는 사진 추가할 때 이미지 피커로 전환되기 전의 엔트리 포인트 역할과, 이미지 피커에서 이미지를 선택한 후 이미지를 표시하기 위함 입니다.
- 현재는 이미지 피커가 존재하지 않기 때문에 초기에는 placeHolder 역할을 하는 아이콘과 텍스트를 띄우고, 버튼을 누르면 이미지 피커 과정을 생략하고 곧바로 imageContent (coil) 을 띄우도록 했습니다.
- coil 을 사용하는 과정에서 Maven Coordinate (라이브러리 명)의 변경과 3.x 의 major release 버전이 최근에 출시되었다는 것을 확인하고 수정했습니다.

## ⏲️Time

    - 3시간

TaewoongR committed:
feat/#48 : Image Icon 추가

- 이미지 선택 버튼에 띄워지는 아이콘

TaewoongR committed:
feat/#48 : CustomImageButton 생성

- 이미지 피커로 연결된 후, 이미지 선택 시 이미지 표현
- 인자 Composable함수의 imageContent 에 coil 함수를 대입해야 함

TaewoongR committed:
feat/#48 : IconSize 추가

TaewoongR committed:
feat/#48 : Custom Button 함수 인자 추가

- showImage를 통해 버튼 클릭 후 coil을 통해 버튼 내부에 사진 출력

TaewoongR committed:
fix/#60 : coil 이름 및 버전 수정

- 3.x major release version 출시
- 라이브러리 Maven coordinates(이름) 변경

TaewoongR committed:
feat/#48 : Custom ImageButton 테스트

- 추가 화면 생성 및 ImageButton 적용
- 모듈 레벨 gralde에 coil 추가 후 에뮬레이터 테스트

haeti-dev commented on core/designsystem/src/main/kotlin/com/boostcamp/mapisode/designsystem/compose/button/MapisodeImageButton.kt:
```diff
@@ -0,0 +1,66 @@
+import com.boostcamp.mapisode.designsystem.compose.MapisodeText
+import com.boostcamp.mapisode.designsystem.theme.MapisodeTheme
+
+@Composable
+fun MapisodeImageButton(
+	modifier: Modifier = Modifier,
+	onClick: () -> Unit,
+	text: String,
+	showImage: Boolean = true,
+	enabled: Boolean = true,
+	showRipple: Boolean = true,
+	interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+	imageContent: @Composable () -> Unit,
+) {
```
modifier 순서 조정해주시면 감사하겠습니다. 그리고 text는 필수인가요?

TaewoongR replied:
다시 보니까 모든 Button의 modifier 순서를 잘못 설정했었네요. 
onClick, modifier 순으로 설정하겠습니다.

text도 기본값을 넣어서 선택적으로 수정하겠습니다.
기획상으로는 텍스트가 전부 들어가지만 추후에 수정 가능성이 있기에 수정하겠습니다.

haeti-dev commented on gradle/libs.versions.toml:
```diff
@@ -87,8 +87,8 @@ androidx-datastore-core = { group = "androidx.datastore", name = "datastore-core
 # Coil
-coil-compose = { group = "io.coil-kt", name = "coil-compose", version.ref = "coil" }
-coil-network-okhttp = { group = "io.coil-kt", name = "coil-network-okhttp", version.ref = "coil" }
+coil-compose = { group = "io.coil-kt.coil3", name = "coil-compose", version.ref = "coil" }
+coil-network-okhttp = { group = "io.coil-kt.coil3", name = "coil-network-okhttp", version.ref = "coil" }
```
감사합니다 ㅎㅎ

shinythink replied:
새벽에 이것 때문에 삽질했던 거 재미있었어요 ㅎㅎ

haeti-dev reviewed (approved):
고생하셨어요~

shinythink commented on core/designsystem/src/main/kotlin/com/boostcamp/mapisode/designsystem/compose/button/MapisodeImageButton.kt:
```diff
@@ -0,0 +1,66 @@
+		rounding = 8.dp,
+		contentPadding = PaddingValues(16.dp),
+	) {
+		if (showImage) {
```
여기서 말하는 이미지는 뭔가요?? 사용자가 이미지를 선택하면 showImage가 false가 되는건가요?

TaewoongR replied:
이 Boolean은 나중에 이미지 피커를 통해 이미지 Uri가 받아졌는지 아닌지의 플래그 역할입니다.

shinythinking reviewed (approved):
수고 많으셨습니다!
코드 읽으면서 즐거웠습니다 ㅎㅎ

TaewoongR committed:
fix/#48 : Button 수정

- onClick, modifier 순으로 인자 순서 변경
- ImageButton의 text파라미터에 기본값을 부여하여 사용자에게 자유도 제공
